Assets/NuGet/Editor/NugetHelper.cs

```diff
@@ -1291,5 +1302,80 @@ public static Texture2D DownloadImage(string url)
+        /// tool. Downloading it from the VSTS instance if needed.
+        /// </summary>
+        /// <param name="packageHost">The hostname where the VSTS instance is hosted (such as microsoft.pkgs.visualsudio.com</param>
+        /// <returns>The password in the form of a token, or null if the password could not be aquired</returns>
+        private static string GetPasswordFromVSTSCredentialProvider(string packageHost)
+        {
+            string credentialProviderBundleFilename = "CredentialProviderBundle.zip";
+            string credentialProviderFilename = "credentialprovider.vss.exe";
+            string credentialProviderPath = Path.Combine(NugetConfigFile.RepositoryPath, credentialProviderFilename);
+
+            // Download CredentialProvider if it's not already installed
+            if (!File.Exists(credentialProviderPath))
+            {
+                HttpWebRequest credentialProviderRequest = (HttpWebRequest)WebRequest.Create("https://" + packageHost + "/_apis/public/nuget/client/CredentialProviderBundle.zip");
+                Stream credentialProviderDownloadStream = credentialProviderRequest.GetResponse().GetResponseStream();
+
+                using (Stream file = File.Create(credentialProviderBundleFilename))
+                {
+                    CopyStream(credentialProviderDownloadStream, file);
+                }
+
+                // Unzip the bundle and extract thje credential provider executable
+                using (ZipFile zip = ZipFile.Read(credentialProviderBundleFilename))
+                {
+                    foreach (ZipEntry entry in zip)
+                    {
+                        if (entry.FileName.ToLower() == credentialProviderFilename)
+                        {
+                            entry.Extract(NugetConfigFile.RepositoryPath, ExtractExistingFileAction.OverwriteSilently);
+                        }
+                    }
+                }
+
+                // Delete the bundle
+                File.Delete(credentialProviderBundleFilename);
+            }
+
+            // Launch the credential provider executable and get the json encoded response from the std output
+            Process process = new Process();
+            process.StartInfo.UseShellExecute = false;
+            process.StartInfo.CreateNoWindow = true;
+            process.StartInfo.RedirectStandardOutput = true;
+            process.StartInfo.FileName = credentialProviderPath;
+            process.StartInfo.Arguments = "-uri " + "https://" + packageHost;
+
+            // http://stackoverflow.com/questions/16803748/how-to-decode-cmd-output-correctly
+            // Default = 65533, ASCII = ?, Unicode = nothing works at all, UTF-8 = 65533, UTF-7 = 242 = WORKS!, UTF-32 = nothing works at all
+            process.StartInfo.StandardOutputEncoding = Encoding.GetEncoding(850);
+            process.Start();
+            process.WaitForExit();
```

Comment:
Not sure how long this takes. Maybe we should show "thinking" status through a progress dialog?

Reply:
The credential provider should only get hit at points where the nuget feed itself is being hit. I think most of those already have a progress bar dialog shown.